File: 0069-sqrtx/0069-sqrtx.java
class Solution {
    public int mySqrt(int x) {
        if (x <= 1) return x;
        
        long start = 1l, end = (long)x, mid = 1l;
        while (start <= end){
            mid = start + (end - start) / 2;
            if(mid * mid == x) return (int)mid;
            else if (mid * mid < x) start = mid + 1;
            else end = mid - 1;
        }
        return (int)end;
    }
}

File: 0069-sqrtx/0069-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        if (x <= 1) return x;
        
        int start = 1, end = x;
        long long mid;
        while (start <= end){
            mid = start + (end - start) / 2;
            if(mid * mid == x) return mid;
            else if (mid * mid < x) start = ++mid;
            else end = --mid;
        }
        return end;
    }
};